### PR TITLE
simplify probatus: remove 'old' code, update mkdocs config

### PR DESCRIPTION
**Pull request by ReinierKoops**
Remove a lot of the extra packages which makes maintaining MKDOCS easier.

### CODE REVIEW

**Comment by ReinierKoops** [comment]
Also enables a more modular approach to installing packages.

**Comment by ReinierKoops** [comment]
Possible fix for https://github.com/ing-bank/probatus/issues/227

**Comment by ReinierKoops** [comment]
Depends on: https://github.com/ing-bank/probatus/pull/238

**Comment by ReinierKoops** [comment]
This PR should simplify Probatus by deprecating code that hasn't been updated in a long time and simplify some of its dependencies. 

- bucketing is better done in Feature Engine, Skorecard or Sklearn, etc
- statistical tests and sampling are better done in Statsmodels, scipy, etc
- imputation is better done in Feature Engine, etc

**Comment by ReinierKoops** [comment]
@PaulZhutovsky

**Comment by adri0** [comment]
Wow major refactoring going on! :)

File: docs/index.md [hunk 1]
```diff
@@ -1,5 +1,3 @@
-# Welcome to probatus documentation!
-
 <img src="img/logo_large.png" width=190 align="right">
 
 **Probatus** is a Python library that allows to analyse binary classification models as well as the data used to develop them.
```
Comment (adri0) [inline docs/index.md:3]: Since you're fixing docs here. Maybe we can fix the git clone link below?

Comment (adri0) [reply docs/index.md:3]: Even better. Idk if it's possible, maybe this index could be generated from the README.md when building? It's essentially the same info, but here it's totally outdated.

File: docs/img/autodist.png
Comment (adri0) [inline docs/img/autodist.png:1]: There's also this `KS2_Example.png` that apparently is not being used anywhere.

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -57,12 +56,4 @@ repos:
...
       name: 'pyupgrade: Updates code to Python 3.8+ code convention'
-      args: [*py_version]
-  - repo: local
-    hooks:
-    - id: black
-      name: 'black: PEP8 compliant code formatter'
-      entry: black
-      language: python
-      types: [python]
-      language_version: python3
```
[the remark this answers is not in the document]
Comment (ReinierKoops) [reply .pre-commit-config.yaml:68]: thanks for the review :D integrated via ruff-format

File: CHANGELOG.md
[the remark this answers is not in the document]
Comment (ReinierKoops) [reply CHANGELOG.md:1]: I think that would decrease some of our workload

File: docs/index.md (hunk 1 above)
Comment (ReinierKoops) [reply docs/index.md:3]: Would be great! Do you perhaps have a proposal for that how to do that?

File: CHANGELOG.md
[the remark this answers is not in the document]
Comment (adri0) [reply CHANGELOG.md:1]: Sounds good :)

File: docs/index.md (hunk 1 above)
Comment (adri0) [reply docs/index.md:3]: I did some googling. It [may be not possible](https://github.com/mkdocs/mkdocs/issues/3152). So I guess I'd just keep the very basic and remove the outdated info, adding that the instructions on how to run/build are in the readme.md in the repo's root.

**Review by adri0 — APPROVED** [review]

**Pull request merged**